_studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h

```diff
@@ -906,29 +914,12 @@ inline bool CompareSegmentMaps(mfxExtVP9Segmentation const & first, mfxExtVP9Seg
 
 inline bool CompareSegmentParams(mfxExtVP9Segmentation const & first, mfxExtVP9Segmentation const & second)
 {
-    bool equal = false;
-
-    if (first.NumSegments == second.NumSegments)
-    {
-        mfxU8 equalParams = 0;
-        for (mfxU8 i = 0; i < first.NumSegments; i++)
-        {
-            if (0 == memcmp(&first.Segment[i], &second.Segment[i], sizeof(mfxVP9SegmentParam)))
-            {
-                equalParams++;
```

Comment:
Why would you stop comparing and return FALSE on the very first non-equal param? No need for equalParams var in this case.